plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AwsContainerOptionsMapper.groovy

```diff
@@ -101,25 +104,33 @@ class AwsContainerOptionsMapper {
                 throw new IllegalArgumentException("Found a malformed value '${ovalue}' for --tmpfs option")
             }
         }
-        if ( tmpfs.size() > 0 )
+        if ( tmpfs.size() > 0 ) {
             params.setTmpfs(tmpfs)
+            atLeastOneSet = true
+        }
 
         // swap limit equal to memory plus swap
         value = findOptionWithSingleValue('memory-swap')
-        if ( value && value.length() > 0)
+        if ( value && value.length() > 0) {
             params.setMaxSwap(value as Integer)
+            atLeastOneSet = true
+        }
 
         // run an init inside the container
-        value = findOptionWithBooleanValue('init')
-        if ( value )
-            params.setInitProcessEnabled(value)
+        if ( findOptionWithBooleanValue('init') ) {
+            params.setInitProcessEnabled(true)
+            atLeastOneSet = true
+        }
 
         // tune container memory swappiness
         value = findOptionWithSingleValue('memory-swappiness')
-        if ( value && value.length() > 0 )
+        if ( value && value.length() > 0 ) {
             params.setSwappiness(value as Integer)
+            atLeastOneSet = true
+        }
 
-        containerProperties.setLinuxParameters(params)
+        if ( atLeastOneSet )
```

Comment:
Is there a side-effect setting an empty `LinuxParameters` object? 

Also, it may be better to have more functional behaviour having `checkLinuxParameters` returning the `LinuxParameters`. 

Finally `checkLinuxParameters` could be a static method to prevent creating a class for each task.

Reply:
> Is there a side-effect setting an empty `LinuxParameters` object?
> 

It's null by default. I wanted to keep it that way if there are no parameters set. This way we have the same (super-tested) behavior previous to this PR. If no param is set the object is null.

> Also, it may be better to have more functional behaviour having `checkLinuxParameters` returning the `LinuxParameters`.

It makes sense.
> 
> Finally `checkLinuxParameters` could be a static method to prevent creating a class for each task.

Makes sense as well, I'll change it (eventually, it'd be an object for each task :) )

Reply:
I made the entire AWSContainerMapper class static to avoid creating unnecessary objects for each task.